app/javascript/pages/browse/MapBrowser.vue

```diff
@@ -2,105 +2,131 @@
-      console.log('Map zoomed')
-    },
-    clicked(map, e) {
-      const title = e.features[0].properties.title
-      console.log(title)
-    },
-    geolocateError(control, positionError) {
-      console.log(positionError)
-    },
-    geolocate(control, position) {
-      console.log(
-        `User position: ${position.coords.latitude}, ${position.coords.longitude}`
-      )
-    },
-  },
-}
-</script>
+                  geojson.features = [ ...geojson.features, {
+                    'type': 'Feature',
+                    'properties': {
+                      'title': contribution.name,
+                      'description': contribution.description ? contribution.description : "",
+                      'categoryTag': contribution.category_tags[0].name,
+                      'urgency': contribution.urgency ? contribution.urgency : "",
+                      'icon': 'harbor',
+                      'iconSize': [50, 50],
+                      'contributionType': contribution.contribution_type,
+                    },
+                    'geometry': {
+                      'type': 'Point',
+                      'coordinates': match.features[0].center
+                    }
+                  } ]
+                })
+                .then(() => {
+                  if (forwarded === this.contributions.length) resolve()
+                })
+              })
+            })
+        }
+
+        buildGeoJson()
+          .then(() => {
+            map.addLayer({
+              id: 'points',
+              type: 'symbol',
+              source: {
+                type: 'geojson',
+                data: geojson,
+              },
+              layout: {
+                'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
+                'text-offset': [0, 0.6],
+                'text-anchor': 'top',
+              },
+            })
+
+            geojson.features.forEach(function (marker) {
+              var el = document.createElement('div');
+              el.className = 'marker ' + marker.properties.contributionType.toLowerCase() + '-marker';
+              el.style.width = marker.properties.iconSize[0] + 'px';
+              el.style.height = marker.properties.iconSize[1] + 'px';
 
-<style>
-  #map {
-    width: 100%;
-    height: 500px;
+              const newMarker = new mapboxgl.Marker(el)
+                .setLngLat(marker.geometry.coordinates)
+                .setPopup(new mapboxgl.Popup({ offset: 35 })
+                .setHTML('<h3>' + marker.properties.title + '</h3><p>' + marker.properties.categoryTag + '</p><p>' + marker.properties.description + '</p>'))
+
+              newMarker.addTo(map)
+            })
+          })
+      },
+      zoomend(map, e) {
+        console.log('Map zoomed')
+      },
+      clicked(map, e) {
+        const title = e.features[0].properties.title
+        console.log(title)
+      },
+      geolocateError(control, positionError) {
+        console.log(positionError)
```

Comment:
I think we *might* want to remove these `console.log` statements, but I will defer to what MA desires here.